src/fsinfo.ts

```diff
@@ -103,6 +103,12 @@ export function fsinfo(path: string, attrs: string[], options?: cockpit.JsonObje
             const target = entry?.target;
             if (!target)
                 return entry;
+            if (target === '.')
+                return self.state.info;
+            // HACK: fsinfo should return this, but doesn't.
+            // We know we only need `.type`, so hardcode it.
+            if (target === '..')
+                return { type: 'dir' };
```

Comment:
I expected this to crash and burn but it does not! *shock*

So you do get this funny situation :rofl: if you follow the symlinks (nothing for us to do here)

![image](https://github.com/cockpit-project/cockpit-files/assets/67428/93ea3e83-8926-49bf-a482-4f615e07940d)

The edit permissions dialog changes the target, unsure if that is wanted behavior https://github.com/cockpit-project/cockpit-files/issues/503